plugins/item_licenses/server/constants.py

```diff
@@ -0,0 +1,98 @@
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+###############################################################################
+
+
+class PluginSettings:
+    LICENSES = 'item_licenses.licenses'
+
+
+class PluginSettingsDefaults:
+    defaults = {
+        PluginSettings.LICENSES: [
+            {
+                # "Popular" licenses from
+                # https://opensource.org/licenses/category
+                'category': 'Code Licenses',
+                'licenses': [
+                    {
+                        'name': 'Apache License 2.0'
+                        },
+                    {
+                        'name': 'BSD 3-Clause "New" or "Revised" License'
+                        },
+                    {
+                        'name': 'BSD 2-Clause "Simplified" or "FreeBSD" License'
+                        },
+                    {
+                        'name': 'GNU General Public License (GPL) 2.0'
+                        },
+                    {
+                        'name': 'GNU General Public License (GPL) 3.0'
+                        },
+                    {
+                        'name': 'GNU Lesser General Public License (LGPL) 2.1'
+                        },
+                    {
+                        'name': 'GNU Lesser General Public License (LGPL) 3.0'
+                        },
+                    {
+                        'name': 'MIT License'
+                        },
+                    {
+                        'name': 'Mozilla Public License 2.0'
+                        }
+                    ]
+                },
+            {
+                # See http://creativecommons.org/licenses/
+                'category': 'Content Licenses',
+                'licenses': [
+                    {
+                        'name': 'CC0'
+                        },
+                    {
+                        'name': 'CC Attribution (BY)'
+                        },
+                    {
+                        'name': 'CC Attribution-ShareAlike (BY-SA)'
+                        },
+                    {
+                        'name': 'CC Attribution-NoDerivs (BY-ND)'
+                        },
+                    {
+                        'name': 'CC Attribution-NonCommercial (BY-NC)'
+                        },
+                    {
+                        'name': 'CC Attribution-NonCommercial-ShareAlike '
+                        '(BY-NC-SA)'
+                        },
+                    {
+                        'name': 'CC Attribution-NonCommercial-NoDerivs '
+                        '(BY-NC-ND)'
+                        },
+                    {
+                        'name': 'Public Domain'
```

Comment:
Perhaps next to CC0 or PDDL we could have additional text reading something
like "Use this to waive copyrights and allow full public use", and,
similarly, additional text next to Public Domain reading something like
"Only for works already in the public domain".

On Wed, May 18, 2016 at 3:47 PM, Brian Helba <notifications@github.com>
wrote:

> In plugins/item_licenses/server/constants.py
> <https://github.com/girder/girder/pull/1347#discussion_r63769235>:
>
> > +                    {
> > +                        'name': 'CC Attribution-NoDerivs (BY-ND)'
> > +                        },
> > +                    {
> > +                        'name': 'CC Attribution-NonCommercial (BY-NC)'
> > +                        },
> > +                    {
> > +                        'name': 'CC Attribution-NonCommercial-ShareAlike '
> > +                        '(BY-NC-SA)'
> > +                        },
> > +                    {
> > +                        'name': 'CC Attribution-NonCommercial-NoDerivs '
> > +                        '(BY-NC-ND)'
> > +                        },
> > +                    {
> > +                        'name': 'Public Domain'
>
> I'd say we should add all 3 Open Data Commons licenses
> <http://opendatacommons.org/licenses/> to the list.
>
> The PDDL looks to be attempting the same thing as the CC0. We just need to
> find a way to encourage users to mark their content as either of those, and
> to avoid inappropriately using "Public Domain" unless they really
> understand what it actually indicates.
>
> Of course, from the perspective some data store software, we shouldn't
> care about any of these semantics. As advocates of open source, though, I
> think we should do our part to ensure content is actually made free for as
> many people as possible.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/girder/girder/pull/1347/files/687eaa599b413848381c3b3fbc14b43db4b12292#r63769235>
>



-- 
David Manthey
R&D Engineer
Kitware Inc.
(518) 881-4439